client/runtime.go

```diff
@@ -230,7 +230,7 @@ func (r *Runtime) Submit(operation *runtime.ClientOperation) (interface{}, error
 
 	// TODO: pick appropriate media type
 	cmt := r.DefaultMediaType
-	if len(operation.ConsumesMediaTypes) > 0 {
+	if len(operation.ConsumesMediaTypes) > 0 && operation.ConsumesMediaTypes[0] != "" {
```

Comment:
Sure, that's even better. I wasn't quite sure what the TODO comment above would include exactly.